v2/internal/runner/inputs.go

```diff
@@ -0,0 +1,83 @@
+package runner
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/projectdiscovery/gologger"
+	"github.com/projectdiscovery/hmap/store/hybrid"
+	"github.com/projectdiscovery/nuclei/v2/pkg/protocols/http/httpclientpool"
+	"github.com/projectdiscovery/retryablehttp-go"
+	"github.com/remeh/sizedwaitgroup"
+)
+
+const probeBulkSize = 50
+
+// initializeTemplatesHTTPInput initializes the http form of input
+// for any loaded http templates if input is in non-standard format.
+func (r *Runner) initializeTemplatesHTTPInput() (*hybrid.HybridMap, error) {
+	hm, err := hybrid.New(hybrid.DefaultDiskOptions)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create temporary input file")
+	}
+
+	httpclient, err := httpclientpool.Get(r.options, &httpclientpool.Configuration{})
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get http client")
+	}
+	gologger.Info().Msgf("Started probing of non-http URLs")
+
+	var bulkSize = probeBulkSize
+	if r.options.BulkSize > probeBulkSize {
+		bulkSize = r.options.BulkSize
+	}
+	// Probe the non-standard URLs and store them in cache
+	swg := sizedwaitgroup.New(bulkSize)
+	count := 0
+	r.hmapInputProvider.Scan(func(value string) bool {
+		if strings.HasPrefix(value, "http://") || strings.HasPrefix(value, "https://") {
+			return true
+		}
+
+		swg.Add()
+		go func(input string) {
+			defer swg.Done()
+
+			if result := probeURL(input, httpclient); result != "" {
+				count++
```

Comment:
Not sure if this is thread-safe, what about using `atomic.AddIntxxx`?

Reply:
done